syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";

package protarrow.protos;

message MyProto {
  string name = 1;
  int32 id = 2;
  repeated int32 values = 3;
}


message NullableExample {
  message NestedNullableExample {
    int32 nested_int = 1;
  }

  int32 int_value = 1;
  repeated int32 int_repeated = 2;
  map<string, int32> int_map = 3;

  NestedNullableExample message_value = 10;
  repeated NestedNullableExample message_repeated = 11;
  map<string, NestedNullableExample> message_map = 12;
}

message EmptyMessage {
  google.protobuf.Empty empty_value = 1;
}

message NestedEmptyMessage {
  EmptyMessage empty_message = 1;
  repeated EmptyMessage repeated_empty_message = 2;
  map<int32, EmptyMessage> empty_example_message_int32_map = 4;
  map<string, EmptyMessage> empty_example_message_string_map = 5;
}

message MessageWithOptional {
  optional string optional_string = 1;
  string plain_string = 2;
  google.protobuf.StringValue string_value = 3;
  repeated google.protobuf.StringValue string_values = 4;

}
